Test/ServerTelemetryChannel.Test/Shared.Tests/Implementation/BackoffLogicManagerTest.cs

```diff
@@ -0,0 +1,248 @@
+﻿namespace Microsoft.ApplicationInsights.WindowsServer.Channel.Implementation
+{
+    using System;
+#if !NET40
+    using System.Diagnostics.Tracing;
+#endif
+
+    using System.Globalization;
+    using System.Net;
+    using System.Linq;
+
+    using Microsoft.ApplicationInsights.WindowsServer.Channel.Helpers;
+
+#if NET40
+    using Microsoft.Diagnostics.Tracing;
+#endif
+
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+    using Microsoft.ApplicationInsights.Channel;
+    using Microsoft.ApplicationInsights.Channel.Implementation;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation;
+    using Microsoft.ApplicationInsights.Web.TestFramework;
+    using Microsoft.ApplicationInsights.WindowsServer.TelemetryChannel.Helpers;
+    using Microsoft.ApplicationInsights.WindowsServer.TelemetryChannel.Implementation;
+    
+    using Assert = Xunit.Assert;
+
+    public class BackoffLogicManagerTest
+    {
+        [TestClass]
+        public class GetBackendResponse
```

Comment:
We have a convention to create one test class for one product class and name of test class is name of product test class + "Test". Nested classes help organize what is being tested. For example for BackoffLogicManagerTest@GetBackendResponse tests all scenarios involving  BackoffLogicManager.GetBackendResponse 

---
In reply to: [69619157](https://github.com/Microsoft/ApplicationInsights-dotnet/pull/248#discussion_r69619157) [](ancestors = 69619157)